test/functional/test_framework/wallet.py

```diff
@@ -319,6 +309,20 @@ def create_self_transfer(self, *, fee_rate=Decimal("0.003"), fee=Decimal("0"), u
 
         return {"txid": tx["txid"], "wtxid": tx["tx"].getwtxid(), "hex": tx["hex"], "tx": tx["tx"], "new_utxo": tx["new_utxos"][0]}
 
+    def signrawtransactionwithwallet(self, *, tx):
```

Comment:
Any reason for a duplicate method when `sign_tx` already exists?

Reply:
`sign_tx` only works for `MiniWalletMode.RAW_P2PK`, and the default wallet mode is `MiniWalletMode.ADDRESS_OP_TRUE`, if we change `tx.vin` size we need `signrawtransactionwithwallet` to sign raw transaction on mode `MiniWalletMode.RAW_OP_TRUE` or `MiniWalletMode.ADDRESS_OP_TRUE`.

Reply:
I mean you could modify sign_tx instead of adding a new method?